### DISCUSSION TITLE
The difference between tuple[()] and tuple[Never]?

### DISCUSSION THREAD

**Discussion by 9en9i** [microsoft/pyright#7521] [Q&A] [answered]
Hi, I noticed that in pyright `tuple[()]` and `tuple[Never]` are not equivalent. That is, any tuple can be assigned to a variable with an annotation like `tuple[Never]`, not just an empty one. Is this an intentional behavior?

**Comment by erictraut** [comment, marked as answer]
These are two different types according to the typing spec. The type `tuple[()]` is a tuple that has zero entries. The type `tuple[Never]` is a tuple that has one entry whose type is `Never`. The latter can never be constructed, so it's not a very useful type, so I wouldn't expect to see it in code. The former can be constructed and is very useful.

**Reply by 9en9i** [reply]
Thank you for the incredibly quick response!
